source/common/router/config_impl.cc

```diff
@@ -854,5 +876,27 @@ ConfigImpl::ConfigImpl(const envoy::api::v2::RouteConfiguration& config, Runtime
                                                      config.response_headers_to_remove());
 }
 
+PerFilterConfigs::PerFilterConfigs(const Protobuf::Map<std::string, ProtobufWkt::Struct>& configs) {
+  for (const auto& cfg : configs) {
+    const std::string& name = cfg.first;
+    const ProtobufWkt::Struct& struct_config = cfg.second;
+
+    auto& factory = Envoy::Config::Utility::getAndCheckFactory<
+        Server::Configuration::NamedHttpFilterConfigFactory>(name);
+
+    configs_[name] = Envoy::Config::Utility::translateToFactoryRDSConfig(struct_config, factory);
```

Comment:
Need to perform validation when PerFilterConfigs are loaded too.

Reply:
I don't think we can do this unfortunately. What type would we validate? (This is related to me asking very nicely for `Validate()` to be a method on the base proto message which is virtual and empty and can be overriden...

Reply:
Pre-processing is being treated as an additional feature. That will need more thinking. We won't be doing that as part of this PR.